linux/libdriver/base/buf.h

```diff
@@ -0,0 +1,72 @@
+/** @file
+ * @defgroup libevm_buf buf
+ * slice of contiguous memory
+ *
+ * @ingroup libevm
+ * @{ */
+#ifndef EVM_BUF_H
+#define EVM_BUF_H
+#include<stdint.h>
+#include<stddef.h>
+
+/** A slice of contiguous memory from @b p to @b q.
+ * Size can be taken with: `q - p`. */
+typedef struct {
+	uint8_t *p; /**< start of memory region */
```

Comment:
Please don't use `end` because it is a reserved keyword for example in Lua, and what if someone wants to bind this to Lua, although this file is to low level to be exposed in Lua. You prefix with something or could use `start` and `finish` instead.  Just be careful with names that are reserved keywords in other languages.

Reply:
If it was in Lua, it would be a field in a table, where `end` is not reserved.  It’s a bit more awkward to use, but it would still work. I have no idea what the reserved words are in Go or Rust. Anyway, start and finish is better than p and q. ;)

Reply:
Ok, changing it to `begin`, `end` pair.

Reply:
Thinking again, C++ `end` is inclusive right? While this implementation is not.
So using the `begin` & `end` names may actually be misleading.